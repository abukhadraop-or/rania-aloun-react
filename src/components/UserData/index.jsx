import React from 'react';
import PropTypes from 'prop-types';
import smile from 'assets/smile.jpg';
import {
  UserImg,
  UserName,
  Container,
  PublishDate,
  InnerContainer,
} from './user-data.styles';

/**
 * Component showing the publisher data in the article card header.
 *
 * @param {Object} props Props passed to UserData Container.
 * @param {string} props.userName Name of article publisher.
 * @param {string} props.publishDate Date of publishing the article.
 *
 * @return {JSX.Element} Container holding the passed article's publisher data.
 */
function UserData({ userName, publishDate }) {
  return (
    <Container>
      <UserImg src={smile} />
      <InnerContainer>
        <UserName>{userName}</UserName>
        <PublishDate>{publishDate}</PublishDate>
      </InnerContainer>
    </Container>
  );
}

UserData.propTypes = {
  userName: PropTypes.string.isRequired,
  publishDate: PropTypes.string.isRequired,
};

export default UserData;
