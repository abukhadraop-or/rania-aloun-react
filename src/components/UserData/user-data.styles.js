import styled from 'styled-components';
import smile from 'assets/smile.jpg';
import colors from 'styles/colors';

export const Container = styled.div`
  display: flex;
  width: fit-content;
`;

export const UserImg = styled.div`
  background-image: url(${smile});
  background-size: contain;
  border-radius: 50%;
  height: 2rem;
  width: 2rem;
`;

export const InnerContainer = styled.div`
  margin: 0 0 0 0.5rem;
`;

export const UserName = styled.p`
  color: ${colors.boringGreen};
  cursor: pointer;
  font-family: 'source sans pro', sans-serif;
  font-size: 1rem;
  font-weight: 600;
  margin: 0;
  &:hover {
    text-decoration: underline;
  }
`;

export const PublishDate = styled.p`
  color: ${colors.ashGrey};
  font-size: 0.8rem;
  margin: 0;
`;
