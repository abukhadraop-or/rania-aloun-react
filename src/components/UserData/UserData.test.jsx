import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import UserData from '.';

describe('<UserData />', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(
      <UserData userName="Rania" publishDate="18/10/2020" />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
