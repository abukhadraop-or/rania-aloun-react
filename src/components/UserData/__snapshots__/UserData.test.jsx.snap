// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserData /> matches snapshot 1`] = `
<styled.div>
  <styled.img
    src="smile.jpg"
  />
  <styled.div>
    <styled.p>
      Rania
    </styled.p>
    <styled.p>
      18/10/2020
    </styled.p>
  </styled.div>
</styled.div>
`;
